### PR TITLE
Embedded Elixir Expressions in Formatter Plugins

### PR DESCRIPTION
**Pull request by maennchen**
Support Sigils inside embedded expressions in files formatted by a formatter plugin.
Implementation of https://groups.google.com/g/elixir-lang-core/c/jFswzSUjsP4

### CODE REVIEW

**Comment by maennchen** [comment]
@josevalim When I opened the mailing list entry, I didn't fully understand how the formatter handles sigils. It's awesome that this change was that easy.

File: lib/mix/test/mix/tasks/format_test.exs [hunk 1]
```diff
@@ -281,6 +281,45 @@ defmodule Mix.Tasks.FormatTest do
     end)
   end
 
+  defmodule Elixir.ExtensionSfacePlugin do
```
Comment (josevalim) [inline lib/mix/test/mix/tasks/format_test.exs:284]: I don't think we need this new module+case. Instead, you can check in the previous plugins that they are receiving the proper `:sigils` option in their `format/2`. WDYT?

Comment (maennchen) [reply lib/mix/test/mix/tasks/format_test.exs:284]: @josevalim Sure, on it :)

**Comment by maennchen** [comment]
@josevalim

I pushed a version with just a new `assert`.

Completely unrelated issue:

When running `make test_mix` repeatedly, I get this error on the 2nd+ runs. A `make clean`  and `make test_mix` makes the test pass again once.

```
$ make test_mix 
==> mix (ex_unit)
Excluding tags: [windows: true]

....................................................

  1) test runs in daemon mode (Mix.Tasks.ReleaseTest)
     /home/maennchen/Development/open-source/elixir/lib/mix/test/mix/tasks/release_test.exs:686
     ** (ExUnit.TimeoutError) test timed out after 60000ms. You can change the timeout:
     
       1. per test by setting "@tag timeout: x" (accepts :infinity)
       2. per test module by setting "@moduletag timeout: x" (accepts :infinity)
       3. globally via "ExUnit.start(timeout: x)" configuration
       4. by running "mix test --timeout x" which sets timeout
       5. or by running "mix test --trace" which sets timeout to infinity
          (useful when using IEx.pry/0)
     
     where "x" is the timeout given as integer in milliseconds (defaults to 60_000).
     
     stacktrace:
       (stdlib 3.16.1) gen.erl:214: :gen.do_call/4
       (stdlib 3.16.1) gen_server.erl:243: :gen_server.call/3
       (kernel 8.1.2) file.erl:1606: :file.call/2
       test/mix/tasks/release_test.exs:766: anonymous fn/1 in Mix.Tasks.ReleaseTest.wait_until_decoded/1
       test/mix/tasks/release_test.exs:774: Mix.Tasks.ReleaseTest.wait_until/1
       test/mix/tasks/release_test.exs:713: anonymous fn/1 in Mix.Tasks.ReleaseTest."test runs in daemon mode"/1
       (mix 1.14.0-dev) lib/mix/project.ex:396: Mix.Project.in_project/4
       (elixir 1.14.0-dev) lib/file.ex:1555: File.cd!/2
       test/test_helper.exs:127: MixTest.Case.in_fixture/3
       test/mix/tasks/release_test.exs:687: (test)
       (ex_unit 1.14.0-dev) lib/ex_unit/runner.ex:500: ExUnit.Runner.exec_test/1
       (stdlib 3.16.1) timer.erl:166: :timer.tc/1
       (ex_unit 1.14.0-dev) lib/ex_unit/runner.ex:451: anonymous fn/4 in ExUnit.Runner.spawn_test_monitor/4
```

Am I doing something wrong or is there a problem with that specific test?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**